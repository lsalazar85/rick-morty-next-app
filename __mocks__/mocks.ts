import {FILTER_CHARACTER_BY_NAME, GET_ALL_CHARACTERS} from "../queries";

export const mockCharacter = {
    id: 1,
    name: 'Rick Sanchez',
    image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
    status: 'Dead',
    dimensionName: {
        dimension: 'C-333'
    },
    species: 'Human',
    gender: 'Male',
    origin: {
        name: 'Earth'
    },
    location: {
        name: 'Earth'
    },
    episode: [
        {
            name: 'Rick',
            air_date: 'December',
            episode: '1',
            characters: [
                {
                    name: 'Rick'
                }
            ]
        }
    ]
}

export const mockContent = {
    loading: false,
    data: {
        characters: {
            results: [mockCharacter],
            info: {
                count: 90,
                pages: 2
            }
        }
    }
}

export const mocks = [
    {
        request: {
            query: GET_ALL_CHARACTERS,
            variables: {
                page: 1,
            },
        },
        result: {
            data: {
                characters: {
                    info: {
                        count: 671,
                        pages: 34
                    },
                    results: [
                        {
                            id: 1,
                            image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
                            name: "Rick Sanchez",
                            dimension: {
                                dimension: "Replacement Dimension",
                                name: "Earth (Replacement Dimension)",
                                type: "Planet"
                            },
                            status: "Alive",
                            species: "Human",
                            gender: "Male",
                            origin: {
                                name: "Earth (C-137)"
                            },
                            location: {
                                name: "Earth (Replacement Dimension)"
                            },
                            episode: [
                                {
                                    id: "1",
                                    name: "Pilot",
                                    air_date: "December 2, 2013"
                                }
                            ]
                        }
                    ]
                }
            },
        },
    },
    {
        request: {
            query: FILTER_CHARACTER_BY_NAME,
            variables: {
                page: 1,
                character: 'Rick'
            },
        },
        result: {
            data: {
                characters: {
                    info: {
                        count: 671,
                        pages: 34
                    },
                    results: [
                        {
                            id: 1,
                            image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
                            name: "Rick Sanchez",
                            dimension: {
                                dimension: "Replacement Dimension",
                                name: "Earth (Replacement Dimension)",
                                type: "Planet"
                            },
                            status: "Alive",
                            species: "Human",
                            gender: "Male",
                            origin: {
                                name: "Earth (C-137)"
                            },
                            location: {
                                name: "Earth (Replacement Dimension)"
                            },
                            episode: [
                                {
                                    id: "1",
                                    name: "Pilot",
                                    air_date: "December 2, 2013"
                                }
                            ]
                        }
                    ]
                }
            },
        },
    },
];
