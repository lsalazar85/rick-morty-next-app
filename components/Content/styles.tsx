import styled from "styled-components";
import colors from "../../styles/colors";

export const ContentContainer = styled.section`
  height: 100%;
  display: flex;
  flex-direction: column;
  padding: 1rem;
  overflow-y: scroll;
  
  ::-webkit-scrollbar {
    width: 0.1rem;
    background: ${colors.Black};
  }

  ::-webkit-scrollbar-thumb {
    background: ${colors.MainGreen};
  }

  @media(min-width: 320px){
    width: 100%;
  }

  @media(min-width: 1024px){
    width: 1024px;
  }
`