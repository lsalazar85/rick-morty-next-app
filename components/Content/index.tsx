import Loader from "../Loader";
import DataNotFound from "../DataNotFound";
import Card from "../Card";
import {ContentProps} from '../../interfaces'

import { ContentContainer } from "./styles";
import { CardSection } from "../../styles/styles";

interface Props {
    dataObject: ContentProps
}

const Content = ({ dataObject }: Props) => {
    const { loading, error, data } = dataObject

    return (
        <ContentContainer>
            {loading && <Loader />}
            {error && <DataNotFound message="Character Not Found" />}
            <CardSection>
                {data?.characters?.results?.map((character:any) => (
                    <Card key={character.id} character={character} />
                ))}
            </CardSection>
        </ContentContainer>
    )
}

export default Content