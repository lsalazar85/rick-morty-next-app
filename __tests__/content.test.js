/**
 * @jest-environment jsdom
 */

import { render  } from '@testing-library/react'
import Content from "@/components/Content";
import {characterObject } from "../__mocks__/mocks";

describe('Content', () => {
    it('should render successfully', () => {
        const { asFragment } = render(<Content dataObject={characterObject}/>)

        expect(asFragment()).toMatchSnapshot();
        expect(asFragment()).toBeDefined()
    })
})