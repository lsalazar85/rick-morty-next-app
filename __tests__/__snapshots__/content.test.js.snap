// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render successfully 1`] = `
<DocumentFragment>
  <section
    class="styles__ContentContainer-sc-773toi-0 dKzdKd"
  >
    <div
      class="styles__CardSection-sc-1d2ydxg-1 kIIQyV"
    />
  </section>
</DocumentFragment>
`;
